From stable-bounces@linux.kernel.org  Tue Jun 20 00:32:35 2006
Date: Tue, 20 Jun 2006 00:32:01 -0700
From: akpm@osdl.org
To: kirr@mns.spb.ru, stable@kernel.org
Cc: 
Subject: x86: compile fix for asm-i386/alternatives.h

From: Kirill Smelkov <kirr@mns.spb.ru>

compile fix:  <asm-i386/alternative.h>  needs  <asm/types.h> for 'u8' --
just look at struct alt_instr.

My module includes <asm/bitops.h> as the first header, and as of 2.6.17 this
leads to compilation errors.

Signed-off-by: Kirill Smelkov <kirr@mns.spb.ru>
Cc: <stable@kernel.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 include/asm-i386/alternative.h |    2 ++
 1 file changed, 2 insertions(+)

--- linux-2.6.17.1.orig/include/asm-i386/alternative.h
+++ linux-2.6.17.1/include/asm-i386/alternative.h
@@ -3,6 +3,8 @@
 
 #ifdef __KERNEL__
 
+#include <asm/types.h>
+
 struct alt_instr {
 	u8 *instr; 		/* original instruction */
 	u8 *replacement;
